warehouse/sessions.py

```diff
@@ -145,6 +146,15 @@ def record_auth_timestamp(self):
         self[self._reauth_timestamp_key] = datetime.datetime.now().timestamp()
         self.changed()
 
+    def record_password_timestamp(self, timestamp):
+        self[self._password_timestamp_key] = timestamp
+        self.changed()
+
+    def password_outdated(self, current_password_timestamp):
+        stored_password_timestamp = self.get(self._password_timestamp_key, 0)
+
+        return current_password_timestamp != stored_password_timestamp
```

Comment:
I think this would have the effect of logging out all currently active sessions because their sessions lack the `_password_timestamp` key, and thus this comparison will fail for them.

Reply:
Great catch. I was initially going to implement the password date check based on a >=, then realized equality was a better check.

I'll update the default to None and add a conditional.

Reply:
fixed in https://github.com/pypa/warehouse/pull/10859/commits/61ce40b0b30cadf1a907af1d552299ad474015fd